examples/java/src/main/java/org/apache/beam/examples/common/WriteOneFilePerWindow.java

```diff
@@ -83,31 +73,38 @@ public PDone expand(PCollection<String> input) {
    */
   public static class PerWindowFiles extends FilenamePolicy {
 
-    private final String prefix;
+    private final ResourceId baseFilename;
 
-    public PerWindowFiles(String prefix) {
-      this.prefix = prefix;
+    public PerWindowFiles(ResourceId baseFilename) {
```

Comment:
Tried - didn't simplify caller code (easier to extract directory from ResourceId)